features/bootstrap/Kodify/BlogBundle/Features/Context/FeatureContext.php

```diff
@@ -0,0 +1,234 @@
+            $author = new Author();
+            $author->setName($row['name']);
+            $entityManager->persist($author);
+        }
+
+        $entityManager->flush();
+    }
+
+    /**
+     * @Given the following posts exist:
+     */
+    public function theFollowingPostsExist(TableNode $table)
+    {
+        $entityManager = $this->getEntityManager();
+
+        foreach ($table as $row) {
+            $author = $this->getAuthorByName($row['author']);
+
+            $post = new Post();
+            $post->setTitle($row['title']);
+            $post->setContent($row['content']);
+            $post->setAuthor($author);
+
+            $entityManager->persist($post);
+        }
+
+        $entityManager->flush();
+    }
+
+    /**
+     * @Given I visit the home page
+     */
+    public function iVisitTheHomePage()
+    {
+        $this->visit('/');
+    }
+
+    /**
+     * @Then The post with title :title is on first column, first row
+     */
+    public function thePostWithTitleIsOnFirstColumnFirstRow($title)
+    {
+        $this->assertPageContainsText($title);
+    }
+
+    /**
+     * @Then The post with title :title is on the second column, first row
+     */
+    public function thePostWithTitleIsOnTheSecondColumnFirstRow($title)
+    {
+        $this->assertPageContainsText($title);
+    }
+
+    /**
+     * @Then The post with title :title is on the first column, second row
+     */
+    public function thePostWithTitleIsOnTheFirstColumnSecondRow($title)
+    {
+        $this->assertPageContainsText($title);
+    }
+
+    /**
+     * @Given the following comments exist:
+     */
+    public function theFollowingCommentsExist(TableNode $table)
+    {
+        $entityManager = $this->getEntityManager();
+
+        foreach ($table as $row) {
+            $author = $this->getAuthorByName($row['author']);
+            $post = $this->getPostByTitle($row['post title']);
+
+            $comment= new Comment();
+            $comment->setAuthor($author);
+            $comment->setPost($post);
+            $comment->setContent($row['text']);
+
+            $entityManager->persist($comment);
+        }
+
+        $entityManager->flush();
+    }
+
+    /**
+     * @Given I visit the page for the post with title :title
+     */
+    public function iVisitThePageForThePostWithTitle($title)
+    {
+        $entityManager = $this->getEntityManager();
+
+        $post = $this->getPostByTitle($title, $entityManager);
+
+        $url = '/posts/' . $post->getId();
+
+        $this->visit($url);
+    }
+
+    /**
+     * @Then I should see a message saying there are no comments
+     */
+    public function iShouldSeeAMessageSayingThereAreNoComments()
+    {
+        $this->assertPageContainsText('No comments');
+    }
+
+    /**
+     * @Then I should see a comments section with :arg1 comment
+     */
+    public function iShouldSeeACommentsSectionWithComment($content)
+    {
+        $this->assertPageNotContainsText('No comments');
```

Comment:
Yes, I think so. I was in a hurry when I realised that I had wasted a lot of time.